src/scitacean/client.py

```diff
@@ -829,6 +828,30 @@ def create_attachment_for_dataset(
             model.DownloadAttachment, _strict_validation=False, **uploaded
         )
 
+    def validate_dataset_model(
+        self, dset: Union[model.UploadDerivedDataset, model.UploadRawDataset]
+    ) -> None:
+        """Validate a dataset in SciCat.
+
+        Parameters
+        ----------
+        dset:
+            Model of the dataset to validate.
+
+        Raises
+        ------
+        ValueError
+            If the dataset does not pass validation.
+        """
+        response = self._call_endpoint(
+            cmd="post",
+            url="datasets/isValid",
+            data=dset,
+            operation="validate_dataset_model",
+        )
+        if not response["valid"]:
+            raise ValueError(f"Dataset {dset} did not pass validation in SciCat.")
```

Comment:
Does SciCat return details about what fields failed validation? It does when you try to upload. So it would be good to show those here as well.

Reply:
It does not, we checked but it only returns `True` or `False`. Yes, extra info on what part of the validation failed would be nice.

Reply:
Ok

Reply:
Is that a feature we should request from Scicat?

Reply:
A way to make either all uploads (dataset, datablocks, attachments, files) succeed or all fail so we don't end up with partially uploaded data. `create_new_dataset_now` attempts to work in this way but is limited because of the SciCat API. I'm hoping to get a feature that lets us do this better. I's too complicated to explain here, though.